packages/service-core/src/metrics/open-telemetry/util.ts

```diff
@@ -43,31 +39,34 @@ export function createOpenTelemetryMetricsFactory(context: ServiceContext): Metr
     configuredExporters.push(periodicExporter);
   }
 
-  let resolvedMetadata: (metadata: RuntimeMetadata) => void;
-  const runtimeMetadata: Promise<RuntimeMetadata> = new Promise((resolve) => {
-    resolvedMetadata = resolve;
+  let resolvedInstanceId: (id: string) => void;
+  const instanceIdPromise = new Promise<string>((resolve) => {
+    resolvedInstanceId = resolve;
   });
 
   lifeCycleEngine.withLifecycle(null, {
     start: async () => {
       const bucketStorage = storageEngine.activeBucketStorage;
       try {
         const instanceId = await bucketStorage.getPowerSyncInstanceId();
-        resolvedMetadata({ ['instance_id']: instanceId });
+        resolvedInstanceId(instanceId);
       } catch (err) {
-        resolvedMetadata({ ['instance_id']: 'Unknown' });
+        resolvedInstanceId('Unknown');
       }
     }
   });
 
+  const resource = resourceFromAttributes({
+    ['service']: 'PowerSync',
+    ['service.version']: pkg.version,
+    ['instance_id']: instanceIdPromise
+  });
+
+  resource.waitForAsyncAttributes?.();
```

Comment:
Does triggering this method kick-off something in the background? Just wondering why we call this when we don't (and can't since this is a sync function) await it here.